packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/library/UpdateLibraryHandler.java

```diff
@@ -0,0 +1,249 @@
+
+    private final ICollaborativeMessageService messageService;
+
+    private final Counter counter;
+
+    public UpdateLibraryHandler(ILibrarySearchService librarySearchService, ISemanticDataSearchService semanticDataSearchService, ISemanticDataUpdateService semanticDataUpdateService, IEditingContextDependencyLoader editingContextDependencyLoader, IIdentityService identityService, ICollaborativeMessageService messageService, MeterRegistry meterRegistry) {
+        this.librarySearchService = Objects.requireNonNull(librarySearchService);
+        this.semanticDataSearchService = Objects.requireNonNull(semanticDataSearchService);
+        this.semanticDataUpdateService = Objects.requireNonNull(semanticDataUpdateService);
+        this.editingContextDependencyLoader = Objects.requireNonNull(editingContextDependencyLoader);
+        this.identityService = Objects.requireNonNull(identityService);
+        this.messageService = Objects.requireNonNull(messageService);
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+    }
+
+    @Override
+    public boolean canHandle(IEditingContext editingContext, IInput input) {
+        return input instanceof UpdateLibraryInput;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IInput input) {
+        this.counter.increment();
+
+        String message = this.messageService.invalidInput(input.getClass().getSimpleName(), UpdateLibraryInput.class.getSimpleName());
+        IPayload payload = new ErrorPayload(input.id(), message);
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, editingContext.getId(), input);
+
+        if (input instanceof UpdateLibraryInput updateLibraryInput) {
+            if (editingContext instanceof EditingContext siriusWebEditingContext) {
+                Optional<Library> optNewLibrary = this.librarySearchService.findById(UUID.fromString(updateLibraryInput.libraryId()));
+                if (optNewLibrary.isPresent()) {
+                    Library newLibrary = optNewLibrary.get();
+                    // Find the version of the library already in the dependencies.
+                    Optional<Library> optOldLibrary = this.getLibraryDependencyWithDifferentVersion(siriusWebEditingContext, newLibrary);
+                    if (optOldLibrary.isPresent()) {
+                        Library oldLibrary = optOldLibrary.get();
+                        List<Resource> oldLibraryResources = this.getAllResourcesFromLibraryRecursiveDependencies(siriusWebEditingContext, oldLibrary);
+                        this.addLibraryDependency(updateLibraryInput, siriusWebEditingContext, newLibrary);
+                        this.editingContextDependencyLoader.loadDependencies(siriusWebEditingContext);
```

Comment:
As we discussed in the import discussion, we need to reload all the dependencies to make sure we don't have duplicated